drafts/draft-ietf-ippm-ioam-data-integrity.xml

```diff
@@ -791,22 +840,27 @@
         this document. Before that, the decapsulating node updates the ICV field
         in the IOAM Integrity Protection header. The Integrity Check Value (ICV)
         is the result of a GMAC operation over the received ICV field and
-        immutable IOAM-Data-Fields added by the decapsulating node. With the
-        received Nonce field provided to GMAC, the decapsulating node performs
-        the following operation:
+        immutable IOAM-Data-Fields added by the decapsulating node. In the case
+        of IOAM-Data-Fields pre-allocated by the encapsulating node, only those
+        updated by the decapsulating node are part of the GMAC operation. With
+        the received Nonce field provided to GMAC, the decapsulating node
+        performs the following operation:
           <list style="bullets">
-            <t>AAD = (ICV field || node's immutable IOAM-Data-Fields)</t>
+            <t>AAD = (ICV field || node's IOAM-Data-Fields)</t>
             <t>ICV = GMAC(AAD)</t>
           </list>
         </t>
 
-        <!--<t>If the decapsulating node does not add any IOAM-Data-Fields (e.g., it
-        only modifies mutable IOAM-Data-Fields or does nothing), then the
-        decapsulating node MUST NOT update the ICV field in the IOAM Integrity
-        Protection header.</t>-->
+        <t>If the decapsulating node does not add any immutable IOAM-Data-Fields
+        (e.g., it only modifies mutable IOAM-Data-Fields or does nothing), and
+        if the decapsulating node does not update any IOAM-Data-Fields
```

Comment:
see above...

... if the decapsulating node, in case the pre-allocated IOAM trace Option-Type is used, does  not update the "node data list" array...